test/utils/ktesting/signals.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ktesting
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+)
+
+var (
+	interruptCtx  context.Context
+	signalChannel chan os.Signal
+)
+
+func init() {
+	// Setting up signals is intentionally done in an init function because
+	// then importing ktesting in a unit or integration test is sufficient
+	// to activate the signal behavior.
+	signalCtx, _ := signal.NotifyContext(context.Background(), os.Interrupt)
```

Comment:
Before we merge I want to test how this affects Ginkgo: it also registers signal handlers.

Reply:
That works fine.